src/SIGV4ConnectionConfig.test.tsx

```diff
@@ -0,0 +1,97 @@
+import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { SIGV4ConnectionConfig } from 'SIGV4ConnectionConfig';
+import { AwsAuthType } from 'types';
+
+const resetWindow = () => {
+  (window as any).grafanaBootData = {
+    settings: {
+      awsAllowedAuthProviders: [AwsAuthType.Credentials, AwsAuthType.Keys],
+      awsAssumeRoleEnabled: true,
+    },
+  };
+};
+
+describe('SIGV4ConnectionConfig', () => {
+  const setup = (onOptionsChange?: () => {}) => {
+    const props: DataSourcePluginOptionsEditorProps<any, any> = {
+      options: {
+        typeName: '',
+        id: 449,
+        uid: 'oIoBsD_Mz',
+        orgId: 1,
+        name: 'Elasticsearch',
+        type: 'elasticsearch',
+        typeLogoUrl: '',
+        access: 'proxy',
+        url: 'http://test.ts',
+        password: '',
+        user: '',
+        database: '',
+        basicAuth: false,
+        basicAuthUser: '',
+        basicAuthPassword: '',
+        withCredentials: false,
+        isDefault: false,
+        jsonData: {
+          esVersion: '7.10.0',
+          includeFrozen: false,
+          logLevelField: '',
+          logMessageField: '',
+          maxConcurrentShardRequests: 5,
+          sigV4AssumeRoleArn: 'arn:test',
+          sigV4Auth: true,
+          sigV4AuthType: 'credentials',
+          sigV4ExternalId: 'test-id',
+          sigV4Profile: 'profile',
+          sigV4Region: 'us-east-2',
+          timeField: '@timestamp',
+        },
+        secureJsonFields: { sigV4AccessKey: true, sigV4SecretKey: true },
+        version: 6,
+        readOnly: false,
+      },
+      onOptionsChange: onOptionsChange ?? jest.fn(),
+    };
+
+    render(<SIGV4ConnectionConfig {...props} />);
+  };
+
+  beforeEach(() => resetWindow());
+  afterEach(() => resetWindow());
+
+  it('should map incoming props correctly', () => {
+    setup();
+    expect(screen.getByText('Credentials file')).toBeInTheDocument();
+    expect(screen.getByLabelText('Assume Role ARN')).toHaveValue('arn:test');
+    expect(screen.getByLabelText('External ID')).toHaveValue('test-id');
+  });
+
+  it('should map changed fields correctly', async () => {
+    const onOptionsChange = jest.fn();
+    setup(onOptionsChange);
```

Comment:
I think I have a preference here to do something like this instead of setup():
```
it('should map incoming props correctly', () => {
  const props = createDefaultProps() <-- the same as your setup call but you return props rather than have a side-effect of rendering
  render(<SIGV4ConnectionConfig {...props} />);
///whatever else the same
// and in tests when you need to override onOptionsChange
  const props = createDefaultProps();
  const onOptionsChange = jest.fn()
  render(<SIGV4ConnectionConfig {...props} onOptionsChange={onOptionsChange} />);
```

I just find it slightly easier to read, but again very nitpicky, please feel free to ignore if you disagree haha